src/lib/db/index.ts

```diff
@@ -0,0 +1,69 @@
+import knex, { Knex } from 'knex'
+import fs from 'fs' 
+import path from 'path'
+
+import type { Logger } from 'pino'
+import { create } from '../logger/index' 
+import env from '../../env'
+
+interface Config {
+  client: 'pg' 
+  migrations: {
+    tableName: 'migrations'
+    directory: string 
+    extension: 'ts' 
+  },
+  connection: {
+    host: string 
+    port: number
+    user: string
+    password: string
+    database: string 
+  },
+}
+
+const config: Config = {
+  client: 'pg',
+  migrations: {
+    tableName: 'migrations',
+    directory: '/migration',
```

Comment:
absolute dir? is this correct?

Reply:
yes, because of typescript and the way we configured to use use `/src` directory. Even tho if it's at the root level causes issues because typescript's rootDir is src/**/*....

Reply:
So as a result kept everything database/storage related in lib/db/ folder